src/ReactiveDomain.Testing/Specifications/TestQueue.cs

```diff
@@ -86,10 +86,12 @@ public void Handle(IMessage message)
             Messages.Enqueue(message);
 
             if (_trackTypes) { _handledTypes.Add(msgType); }
-            if (_idWatchList.Count > 0)
+            lock (_idWatchList)
             {
-                _idWatchList.TryGetValue(message.MsgId, out var mres);
-                mres?.Set();
+                if (_idWatchList.ContainsKey(message.MsgId))
+                {
+                    _idWatchList[message.MsgId].Set();
+                }
```

Comment:
FWIW, the `TryGetValue` code is in principle more efficient since it does only one lookup instead of two.

There are a few changes like that in here.

Reply:
I honestly like the clarity of doing this way in the code

Reply:
and it's a test class, not in a hot path anywhere